src/sql/workbench/contrib/notebook/browser/notebookExplorer/searchResultsViewPane.ts

```diff
@@ -237,7 +241,57 @@ export class NotebookSearchView extends SearchView {
 		}));
 	}
 
-	public startSearch(query: ITextQuery, excludePatternText: string, includePatternText: string, triggeredOnType: boolean, searchWidget: NotebookSearchWidget): Thenable<void> {
+	public async validateAndSearch(query: ITextQuery, searchWidget: SimpleSearchWidget): Promise<void> {
+		try {
+			await this.validateQry(query);
+			if (this.parentContainer.views.length > 1) {
+				let filesToIncludeFiltered: string = '';
+				this.parentContainer.views.forEach(async (v) => {
+					let booksViewPane = (<TreeViewPane>this.parentContainer.getView(v.id));
+					if (booksViewPane?.treeView?.root) {
+						let root = booksViewPane.treeView.root;
+						if (root.children) {
```

Comment:
I'd get rid of the if and consolidate this to one line 

`booksViewPane.treeView.root.children?.forEach(...`